File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/Responsavel.java
package br.com.kronos.backend.aplicacao.pessoa;

import static java.util.Objects.requireNonNull;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Responsavel {
	private int id;
    private LocalDate dataInicio;
    private LocalDate dataFim;
    private int prioridade;
    private Long idPessoa;
    private Long idPessoaResponsavel;
    private Integer idTipoResponsavel;

	@Builder
	public Responsavel(int id, LocalDate dataInicio, LocalDate dataFim, int prioridade, Long idPessoa, Long idPessoaResponsavel, Integer idTipoResponsavel) {

		super();
        this.id = id;
        this.dataInicio = requireNonNull(dataInicio, "dataInicio é obrigatório");
        this.dataFim = dataFim;
        this.prioridade = requireNonNull(prioridade, "prioridade é obrigatório");
        this.idPessoa = requireNonNull(idPessoa, "idPessoa é obrigatório"); 
        this.idPessoaResponsavel = requireNonNull(idPessoaResponsavel, "idPessoaResponsavel é obrigatório");
        this.idTipoResponsavel = requireNonNull(idTipoResponsavel, "idTipoResponsavel é obrigatório");
       
	}
}

File: poc-kronos-frontend/src/model/pessoa/Responsavel.js
class Responsavel {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataInicio = dados ? dados.dataInicio : null;
    this.dataFim = dados ? dados.dataFim : null;
    this.prioridade = dados ? dados.prioridade : null;
    this.idPessoa = dados ? dados.idPessoa : null;
    this.idPessoaResponsavel = dados ? dados.idPessoaResponsavel : null;
    this.nomePessoaResponsavel = dados ? dados.nomePessoaResponsavel : null;
    this.idTipoResponsavel = dados ? dados.idTipoResponsavel : null;
    this.nomeTipoResponsavel = dados ? dados.nomeTipoResponsavel : null;
  }

  static validations(required, dataFimMenor, maxValue, numeric) {
    return {
      dataInicio: { required },
      dataFim: { dataFimMenor: dataFimMenor("dataInicio") },
      prioridade: { required, numeric, maxValue: maxValue(5) },
      idPessoa: { required },
      idPessoaResponsavel: { required },
      idTipoResponsavel: { required }
    };
  }

  static validationMessages() {
    return {
      dataInicio: {
        required: "Data Início é obrigatória",
        dataFimMenor: "Data Fim não pode ser menor que Data Inicial"
      },
      dataFim: {
        dataFimMenor: "Data Fim não pode ser menor que Data Inicial"
      },
      prioridade: {
        required: "Prioridade é obrigatório",
        numeric: "Somente número",
        maxValue: "Valor máximo 5"
      },
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idPessoaResponsavel: {
        required: "Pessoa Responsável é obrigatória"
      },
      idTipoResponsavel: {
        required: "Tipo Responsável é obrigatório"
      }
    };
  }
}
export default Responsavel;
